{
    "active_lock_reason": null,
    "assignee": {
        "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
        "events_url": "https://api.github.com/users/seisman/events{/privacy}",
        "followers_url": "https://api.github.com/users/seisman/followers",
        "following_url": "https://api.github.com/users/seisman/following{/other_user}",
        "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/seisman",
        "id": 3974108,
        "login": "seisman",
        "node_id": "MDQ6VXNlcjM5NzQxMDg=",
        "organizations_url": "https://api.github.com/users/seisman/orgs",
        "received_events_url": "https://api.github.com/users/seisman/received_events",
        "repos_url": "https://api.github.com/users/seisman/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/seisman"
    },
    "assignees": [
        {
            "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
            "events_url": "https://api.github.com/users/seisman/events{/privacy}",
            "followers_url": "https://api.github.com/users/seisman/followers",
            "following_url": "https://api.github.com/users/seisman/following{/other_user}",
            "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/seisman",
            "id": 3974108,
            "login": "seisman",
            "node_id": "MDQ6VXNlcjM5NzQxMDg=",
            "organizations_url": "https://api.github.com/users/seisman/orgs",
            "received_events_url": "https://api.github.com/users/seisman/received_events",
            "repos_url": "https://api.github.com/users/seisman/repos",
            "site_admin": false,
            "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
            "type": "User",
            "url": "https://api.github.com/users/seisman"
        }
    ],
    "author_association": "NONE",
    "body": "I am using pygmt to graph focal mechanisms with magnitude scale.\r\nHowever, when I try to give it an offset so that they don't pile up the function doesn't seem to work.\r\n\r\nI'm using :\r\n\r\n```python\r\nspec = dict (strike=strikes[0], dip=dips[0], rake=rakes[0], magnitude=magnitude [0])\r\n\r\nfig.meca (spec, scale=f \".{0.002 * (2 ** magnitude[0])}c\", longitude = lons [0], latitude = lats [0], depth = 25.0, convention = \"aki\", plot_longitude = lons [0], plot_latitude = lats[0] -10, offset=True) \r\n```\r\nwhere all the objects are lists with real (float) values\r\n\r\n**System information**\r\n```\r\n# Name                    Version                   Build  Channel\r\npygmt                     0.3.1              pyhd8ed1ab_0    conda-forge\r\n\r\n```\r\n\r\nDo I make something wrong?\r\n",
    "closed_at": "2022-07-03T15:56:18Z",
    "closed_by": {
        "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
        "events_url": "https://api.github.com/users/seisman/events{/privacy}",
        "followers_url": "https://api.github.com/users/seisman/followers",
        "following_url": "https://api.github.com/users/seisman/following{/other_user}",
        "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/seisman",
        "id": 3974108,
        "login": "seisman",
        "node_id": "MDQ6VXNlcjM5NzQxMDg=",
        "organizations_url": "https://api.github.com/users/seisman/orgs",
        "received_events_url": "https://api.github.com/users/seisman/received_events",
        "repos_url": "https://api.github.com/users/seisman/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/seisman"
    },
    "comments": 14,
    "comments_content": [
        {
            "author_association": "NONE",
            "body": "\ud83d\udc4b Thanks for opening your first issue here! Please make sure you filled out the template with as much detail as possible. You might also want to take a look at our [contributing guidelines](https://github.com/GenericMappingTools/pygmt/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/GenericMappingTools/pygmt/blob/master/CODE_OF_CONDUCT.md).\n",
            "created_at": "2021-03-26T18:04:34Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1129#issuecomment-808417299",
            "id": 808417299,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129",
            "node_id": "MDEyOklzc3VlQ29tbWVudDgwODQxNzI5OQ==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/808417299/reactions"
            },
            "updated_at": "2021-03-26T18:04:34Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/808417299",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/in/4141?v=4",
                "events_url": "https://api.github.com/users/welcome%5Bbot%5D/events{/privacy}",
                "followers_url": "https://api.github.com/users/welcome%5Bbot%5D/followers",
                "following_url": "https://api.github.com/users/welcome%5Bbot%5D/following{/other_user}",
                "gists_url": "https://api.github.com/users/welcome%5Bbot%5D/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/apps/welcome",
                "id": 30606887,
                "login": "welcome[bot]",
                "node_id": "MDM6Qm90MzA2MDY4ODc=",
                "organizations_url": "https://api.github.com/users/welcome%5Bbot%5D/orgs",
                "received_events_url": "https://api.github.com/users/welcome%5Bbot%5D/received_events",
                "repos_url": "https://api.github.com/users/welcome%5Bbot%5D/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/welcome%5Bbot%5D/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/welcome%5Bbot%5D/subscriptions",
                "type": "Bot",
                "url": "https://api.github.com/users/welcome%5Bbot%5D"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "Thanks, @rdcorona. I think you find a bug in the `meca` method. I can re-produce this potential bug using [Focal mechanisms](https://www.pygmt.org/dev/gallery/seismology/meca.html#sphx-glr-gallery-seismology-meca-py) gallery example. The earthquake is at `longitude=-124.3` and `latitude=48.1` and the beachball should be plotted at `plot_longitude=-124.3` and `plot_latitude=47.5`. However, the beachball is stilled plotted at the epicenter.\r\n\r\n```Python\r\nimport pygmt\r\nfig = pygmt.Figure()\r\nfig.coast(region=[-125, -122, 47, 49], projection=\"M6c\", land=\"grey\", water=\"lightblue\", shorelines=True, frame=\"a\")\r\nfocal_mechanism = dict(strike=330, dip=30, rake=90, magnitude=3)\r\nfig.meca(focal_mechanism, scale=\"1c\", longitude=-124.3, latitude=48.1, depth=12.0, plot_longitude=-124.3, plot_latitude=47.5, offset=True)\r\nfig.show()\r\n```\r\n![meca](https://user-images.githubusercontent.com/50591376/112717373-32942c00-8f27-11eb-934e-d9924d1e983e.png)\r\n",
            "created_at": "2021-03-27T10:07:10Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1129#issuecomment-808705820",
            "id": 808705820,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129",
            "node_id": "MDEyOklzc3VlQ29tbWVudDgwODcwNTgyMA==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/808705820/reactions"
            },
            "updated_at": "2021-03-27T10:10:45Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/808705820",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/50591376?v=4",
                "events_url": "https://api.github.com/users/core-man/events{/privacy}",
                "followers_url": "https://api.github.com/users/core-man/followers",
                "following_url": "https://api.github.com/users/core-man/following{/other_user}",
                "gists_url": "https://api.github.com/users/core-man/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/core-man",
                "id": 50591376,
                "login": "core-man",
                "node_id": "MDQ6VXNlcjUwNTkxMzc2",
                "organizations_url": "https://api.github.com/users/core-man/orgs",
                "received_events_url": "https://api.github.com/users/core-man/received_events",
                "repos_url": "https://api.github.com/users/core-man/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/core-man/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/core-man/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/core-man"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "I can reproduce the issue, and now know why it fails. Unfortunately, there is no workaround now. ",
            "created_at": "2021-04-01T03:30:19Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1129#issuecomment-811611768",
            "id": 811611768,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129",
            "node_id": "MDEyOklzc3VlQ29tbWVudDgxMTYxMTc2OA==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 2,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 2,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/811611768/reactions"
            },
            "updated_at": "2021-04-01T03:30:19Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/811611768",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
                "events_url": "https://api.github.com/users/seisman/events{/privacy}",
                "followers_url": "https://api.github.com/users/seisman/followers",
                "following_url": "https://api.github.com/users/seisman/following{/other_user}",
                "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/seisman",
                "id": 3974108,
                "login": "seisman",
                "node_id": "MDQ6VXNlcjM5NzQxMDg=",
                "organizations_url": "https://api.github.com/users/seisman/orgs",
                "received_events_url": "https://api.github.com/users/seisman/received_events",
                "repos_url": "https://api.github.com/users/seisman/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/seisman"
            }
        },
        {
            "author_association": "NONE",
            "body": "Hello seisman.\r\nSo, we'll have to wait for the 0.4.0 release.\ud83e\udd37\ud83c\udffb\u200d\u2642\ufe0f\r\n\r\nThanks for the reply",
            "created_at": "2021-04-01T03:50:00Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1129#issuecomment-811617628",
            "id": 811617628,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129",
            "node_id": "MDEyOklzc3VlQ29tbWVudDgxMTYxNzYyOA==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/811617628/reactions"
            },
            "updated_at": "2021-04-01T03:50:00Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/811617628",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/81437398?v=4",
                "events_url": "https://api.github.com/users/rdcorona/events{/privacy}",
                "followers_url": "https://api.github.com/users/rdcorona/followers",
                "following_url": "https://api.github.com/users/rdcorona/following{/other_user}",
                "gists_url": "https://api.github.com/users/rdcorona/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/rdcorona",
                "id": 81437398,
                "login": "rdcorona",
                "node_id": "MDQ6VXNlcjgxNDM3Mzk4",
                "organizations_url": "https://api.github.com/users/rdcorona/orgs",
                "received_events_url": "https://api.github.com/users/rdcorona/received_events",
                "repos_url": "https://api.github.com/users/rdcorona/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/rdcorona/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/rdcorona/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/rdcorona"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "Currently, offsetting beachballs only works for files. \r\n\r\nI'll see what I can do to fix it.",
            "created_at": "2021-04-01T03:52:17Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1129#issuecomment-811618241",
            "id": 811618241,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129",
            "node_id": "MDEyOklzc3VlQ29tbWVudDgxMTYxODI0MQ==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/811618241/reactions"
            },
            "updated_at": "2021-04-01T03:52:17Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/811618241",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
                "events_url": "https://api.github.com/users/seisman/events{/privacy}",
                "followers_url": "https://api.github.com/users/seisman/followers",
                "following_url": "https://api.github.com/users/seisman/following{/other_user}",
                "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/seisman",
                "id": 3974108,
                "login": "seisman",
                "node_id": "MDQ6VXNlcjM5NzQxMDg=",
                "organizations_url": "https://api.github.com/users/seisman/orgs",
                "received_events_url": "https://api.github.com/users/seisman/received_events",
                "repos_url": "https://api.github.com/users/seisman/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/seisman"
            }
        },
        {
            "author_association": "NONE",
            "body": "Ok, meanwhile I'll try it with a dataFrame and an ASCII file \r\n\r\nThanks again ",
            "created_at": "2021-04-01T03:58:57Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1129#issuecomment-811620103",
            "id": 811620103,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129",
            "node_id": "MDEyOklzc3VlQ29tbWVudDgxMTYyMDEwMw==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/811620103/reactions"
            },
            "updated_at": "2021-04-01T03:58:57Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/811620103",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/81437398?v=4",
                "events_url": "https://api.github.com/users/rdcorona/events{/privacy}",
                "followers_url": "https://api.github.com/users/rdcorona/followers",
                "following_url": "https://api.github.com/users/rdcorona/following{/other_user}",
                "gists_url": "https://api.github.com/users/rdcorona/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/rdcorona",
                "id": 81437398,
                "login": "rdcorona",
                "node_id": "MDQ6VXNlcjgxNDM3Mzk4",
                "organizations_url": "https://api.github.com/users/rdcorona/orgs",
                "received_events_url": "https://api.github.com/users/rdcorona/received_events",
                "repos_url": "https://api.github.com/users/rdcorona/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/rdcorona/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/rdcorona/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/rdcorona"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "> I can reproduce the issue, and now know why it fails. Unfortunately, there is no workaround now.\r\n\r\nStill a problem on PyGMT v0.5.0 unfortunately. Could you elaborate more on why it fails? I tried debugging this while handling #1611, and couldn't see any obvious issue on the Python side, e.g. the `plot_longitude` and `plot_latitude` variables seem to be appended to `spec` ok here:\r\n\r\nhttps://github.com/GenericMappingTools/pygmt/blob/9e8a7bd81188f87637a7611239db30283f641b39/pygmt/src/meca.py#L330-L337\r\n\r\nOnly thing I can think of is that this is a problem with the `spec` array to GMT virtualfile convention at https://github.com/GenericMappingTools/pygmt/blob/9e8a7bd81188f87637a7611239db30283f641b39/pygmt/src/meca.py#L454 so should this be marked as an upstream GMT API bug then?",
            "created_at": "2021-11-07T01:36:51Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1129#issuecomment-962536779",
            "id": 962536779,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129",
            "node_id": "IC_kwDOBRZfO845XyVL",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/962536779/reactions"
            },
            "updated_at": "2021-11-07T01:36:51Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/962536779",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
                "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
                "followers_url": "https://api.github.com/users/weiji14/followers",
                "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
                "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/weiji14",
                "id": 23487320,
                "login": "weiji14",
                "node_id": "MDQ6VXNlcjIzNDg3MzIw",
                "organizations_url": "https://api.github.com/users/weiji14/orgs",
                "received_events_url": "https://api.github.com/users/weiji14/received_events",
                "repos_url": "https://api.github.com/users/weiji14/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/weiji14"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "The input parameters of the `meca` module are a little complicated. I will take the Aki & Richard convention (**-Sa**) as an example. The input parameters are:\r\n```\r\nlongitude latitude depth strike dip rake magnitude [plot_longitude plot_latitude] [event_name]\r\n```\r\nAs you can see, plot_longitude & plot_latitude and event_name are optional. Thus users can give input parameters in many different ways:\r\n\r\n1. don't give plot longitude&latitude and event_name\r\n2. give plot longitude&latitude but not event_name\r\n3. give event_name but not plot longitude&latitude\r\n\r\nDue to these complications, the first few columns (7 columns for the **-Sa** case) are read as numeric values (`S->data in the GMT source code)`, thus data passed as vectors (`virtualfile_from_vector`) or a matrix (`virtualfile_from_matrix`) can be read correctly. However, because GMT can't know exactly the number of trailing columns and the `event_name` parameter can contain whitespace, all the trailing columns are read as trailing text (`S->text` in the GMT source code). \r\n\r\nSee https://github.com/GenericMappingTools/gmt/blob/3e68b8de02774f2eda0068ea356586d61770f070/src/seis/psmeca.c#L775  and https://github.com/GenericMappingTools/gmt/blob/3e68b8de02774f2eda0068ea356586d61770f070/src/seis/psmeca.c#L824 if you want to read the GMT source code.\r\n\r\nDue to the above reasons, we have to pass the trailing columns (plot longitude&latitude and event_name) using `GMT_Put_Strings`, not `GMT_Put_Matrix` or `GMT_Put_Vectors`. Actually, we once had similar issues with the `text` function (related to https://github.com/GenericMappingTools/pygmt/issues/483, https://github.com/GenericMappingTools/pygmt/pull/520).\r\n\r\nThus, what we need to do is:\r\n\r\n- Replace `lib.virtualfile_from_matrix` with `lib.virtualfile_from_vector`\r\n- Build a string list/vector for `plot_longitude plot_latitude event_name` and pass it to `lib.virtual_from_vector`.",
            "created_at": "2021-11-07T09:46:04Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1129#issuecomment-962572747",
            "id": 962572747,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129",
            "node_id": "IC_kwDOBRZfO845X7HL",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/962572747/reactions"
            },
            "updated_at": "2021-11-07T09:52:35Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/962572747",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
                "events_url": "https://api.github.com/users/seisman/events{/privacy}",
                "followers_url": "https://api.github.com/users/seisman/followers",
                "following_url": "https://api.github.com/users/seisman/following{/other_user}",
                "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/seisman",
                "id": 3974108,
                "login": "seisman",
                "node_id": "MDQ6VXNlcjM5NzQxMDg=",
                "organizations_url": "https://api.github.com/users/seisman/orgs",
                "received_events_url": "https://api.github.com/users/seisman/received_events",
                "repos_url": "https://api.github.com/users/seisman/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/seisman"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "> Thus, what we need to do is:\r\n> \r\n> * Replace `lib.virtualfile_from_matrix` with `lib.virtualfile_from_vector`\r\n\r\nAh I see, maybe we could use `lib.virtualfile_from_data` as with #949? Since it basically passes things to `lib.virtualfile_from_vectors` anyways:\r\n\r\nhttps://github.com/GenericMappingTools/pygmt/blob/4b80f3c269f39917c8bb7ee96c5a253279877024/pygmt/clib/session.py#L1427-L1431\r\n\r\nthough it sounds like we'll need to be careful and ensure that it doesn't get into this case:\r\n\r\nhttps://github.com/GenericMappingTools/pygmt/blob/4b80f3c269f39917c8bb7ee96c5a253279877024/pygmt/clib/session.py#L1454-L1458\r\n\r\nso some very careful refactoring and unit test writing to do!\r\n\r\n> * Build a string list/vector for `plot_longitude plot_latitude event_name` and pass it to `lib.virtual_from_vector`.\r\n\r\nJust to be clear, do you mean that `plot_longitude`, `plot_latitude` and `event_name` is to be passed in as 3 columns or 1 column using `GMT_Put_Strings`?",
            "created_at": "2021-11-07T09:58:34Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1129#issuecomment-962574090",
            "id": 962574090,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129",
            "node_id": "IC_kwDOBRZfO845X7cK",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/962574090/reactions"
            },
            "updated_at": "2021-11-07T09:00:52Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/962574090",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
                "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
                "followers_url": "https://api.github.com/users/weiji14/followers",
                "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
                "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/weiji14",
                "id": 23487320,
                "login": "weiji14",
                "node_id": "MDQ6VXNlcjIzNDg3MzIw",
                "organizations_url": "https://api.github.com/users/weiji14/orgs",
                "received_events_url": "https://api.github.com/users/weiji14/received_events",
                "repos_url": "https://api.github.com/users/weiji14/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/weiji14"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "> Ah I see, maybe we could use `lib.virtualfile_from_data` as with #949?\r\n\r\nYes, ideally we should use `lib.virtualfile_from_data`.\r\n\r\n> Just to be clear, do you mean that `plot_longitude`, `plot_latitude` and `event_name` is to be passed in as 3 columns or 1 column using `GMT_Put_Strings`?\r\n\r\nWe can only call `GMT_Put_Strings` once, so we have to combine the three parameters into one string vector and pass it to `GMT_Put_Strings`. Actually, we already did it in the `virtualfile_from_vector` function. The `virtualfile_from_vector` function doesn't work for `meca`, because plot_longitude and plot_latitude are numeric vectors but must be passed as strings.\r\n\r\nhttps://github.com/GenericMappingTools/pygmt/blob/4b80f3c269f39917c8bb7ee96c5a253279877024/pygmt/clib/session.py#L1159-L1172.\r\n",
            "created_at": "2021-11-07T09:13:33Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1129#issuecomment-962576217",
            "id": 962576217,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129",
            "node_id": "IC_kwDOBRZfO845X79Z",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/962576217/reactions"
            },
            "updated_at": "2021-11-07T09:13:33Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/962576217",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
                "events_url": "https://api.github.com/users/seisman/events{/privacy}",
                "followers_url": "https://api.github.com/users/seisman/followers",
                "following_url": "https://api.github.com/users/seisman/following{/other_user}",
                "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/seisman",
                "id": 3974108,
                "login": "seisman",
                "node_id": "MDQ6VXNlcjM5NzQxMDg=",
                "organizations_url": "https://api.github.com/users/seisman/orgs",
                "received_events_url": "https://api.github.com/users/seisman/received_events",
                "repos_url": "https://api.github.com/users/seisman/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/seisman"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "I don't know what happened to GitHub. \r\n\r\nMy reply to your comment https://github.com/GenericMappingTools/pygmt/issues/1129#issuecomment-962574090 is displayed before your comment at here: https://github.com/GenericMappingTools/pygmt/issues/1129#issuecomment-962576217",
            "created_at": "2021-11-07T09:16:18Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1129#issuecomment-962576593",
            "id": 962576593,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129",
            "node_id": "IC_kwDOBRZfO845X8DR",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 1,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 1,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/962576593/reactions"
            },
            "updated_at": "2021-11-07T09:16:18Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/962576593",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
                "events_url": "https://api.github.com/users/seisman/events{/privacy}",
                "followers_url": "https://api.github.com/users/seisman/followers",
                "following_url": "https://api.github.com/users/seisman/following{/other_user}",
                "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/seisman",
                "id": 3974108,
                "login": "seisman",
                "node_id": "MDQ6VXNlcjM5NzQxMDg=",
                "organizations_url": "https://api.github.com/users/seisman/orgs",
                "received_events_url": "https://api.github.com/users/seisman/received_events",
                "repos_url": "https://api.github.com/users/seisman/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/seisman"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "> I don't know what happened to GitHub.\r\n> \r\n> My reply to your comment [#1129 (comment)](https://github.com/GenericMappingTools/pygmt/issues/1129#issuecomment-962574090) is displayed before your comment at here: [#1129 (comment)](https://github.com/GenericMappingTools/pygmt/issues/1129#issuecomment-962576217)\r\n\r\nYeah I see the same thing, must be time going backwards :laughing:\r\n\r\n> > Ah I see, maybe we could use `lib.virtualfile_from_data` as with #949?\r\n> \r\n> Yes, ideally we should use `lib.virtualfile_from_data`.\r\n> \r\n> > Just to be clear, do you mean that `plot_longitude`, `plot_latitude` and `event_name` is to be passed in as 3 columns or 1 column using `GMT_Put_Strings`?\r\n> \r\n> We can only call `GMT_Put_Strings` once, so we have to combine the three parameters into one string vector and pass it to `GMT_Put_Strings`. Actually, we already did it in the `virtualfile_from_vector` function. The `virtualfile_from_vector` function doesn't work for `meca`, because plot_longitude and plot_latitude are numeric vectors but must be passed as strings.\r\n\r\nOk, I think I got a clear idea on what to do now. We'll need to convert the `plot_longitude` and `plot_latitude` vectors from numerical arrays to string arrays, and then `lib.virtualfile_from_vectors` (via `lib.virtualfile_from_data`) should do the concatenation magic (joining `event_name` if it is present) and call `GMT_Put_Strings` only once on the concatenated string column.\r\n\r\nProbably do 2 separate PRs:\r\n\r\n1. Refactor from using `lib.virtualfile_from_matrix` to `lib.virtualfile_from_data` (which will enable other input data structures like `xarray.Dataset` arrays) and possibly allow for text labels to work on non-file inputs (fixing https://github.com/GenericMappingTools/pygmt/pull/516#issuecomment-657313059).\r\n2. Do the actual bugfix so that `offset` works.",
            "created_at": "2021-11-07T09:27:57Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1129#issuecomment-962578006",
            "id": 962578006,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129",
            "node_id": "IC_kwDOBRZfO845X8ZW",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/962578006/reactions"
            },
            "updated_at": "2021-11-07T09:32:09Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/962578006",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
                "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
                "followers_url": "https://api.github.com/users/weiji14/followers",
                "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
                "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/weiji14",
                "id": 23487320,
                "login": "weiji14",
                "node_id": "MDQ6VXNlcjIzNDg3MzIw",
                "organizations_url": "https://api.github.com/users/weiji14/orgs",
                "received_events_url": "https://api.github.com/users/weiji14/received_events",
                "repos_url": "https://api.github.com/users/weiji14/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/weiji14"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "> Ok, I think I got a clear idea on what to do now. We'll need to convert the `plot_longitude` and `plot_latitude` vectors from numerical arrays to string arrays, and then `lib.virtualfile_from_vectors` (via `lib.virtualfile_from_data`) should do the concatenation magic (joining `event_name` if it is present) and call `GMT_Put_Strings` only once on the concatenated string column.\r\n> \r\n> Probably do 2 separate PRs. 1) Refactor from using `lib.virtualfile_from_matrix` to `lib.virtualfile_from_data` (which will enable other input data structures like `xarray.Dataset` arrays) and 2) Do the actual bugfix so that `offset` works.\r\n\r\nYes!",
            "created_at": "2021-11-07T09:29:25Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1129#issuecomment-962578179",
            "id": 962578179,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129",
            "node_id": "IC_kwDOBRZfO845X8cD",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/962578179/reactions"
            },
            "updated_at": "2021-11-07T09:29:25Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/962578179",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
                "events_url": "https://api.github.com/users/seisman/events{/privacy}",
                "followers_url": "https://api.github.com/users/seisman/followers",
                "following_url": "https://api.github.com/users/seisman/following{/other_user}",
                "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/seisman",
                "id": 3974108,
                "login": "seisman",
                "node_id": "MDQ6VXNlcjM5NzQxMDg=",
                "organizations_url": "https://api.github.com/users/seisman/orgs",
                "received_events_url": "https://api.github.com/users/seisman/received_events",
                "repos_url": "https://api.github.com/users/seisman/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/seisman"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "> > Ok, I think I got a clear idea on what to do now. We'll need to convert the `plot_longitude` and `plot_latitude` vectors from numerical arrays to string arrays, and then `lib.virtualfile_from_vectors` (via `lib.virtualfile_from_data`) should do the concatenation magic (joining `event_name` if it is present) and call `GMT_Put_Strings` only once on the concatenated string column.\r\n> > Probably do 2 separate PRs. 1) Refactor from using `lib.virtualfile_from_matrix` to `lib.virtualfile_from_data` (which will enable other input data structures like `xarray.Dataset` arrays) and 2) Do the actual bugfix so that `offset` works.\r\n> \r\n> Yes!\r\n\r\nOn third thought, maybe 3 separate PRs (because `pygmt.meca` is a beast to refactor...):\r\n\r\n1. Refactor `meca` to use `virtualfile_from_data` (doing at #1613)\r\n2. Handle text labels and angle/font/justify (c.f. #1466)\r\n3. Fix this `offset` parameter bug (#1129).",
            "created_at": "2021-11-07T11:21:56Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1129#issuecomment-962592735",
            "id": 962592735,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129",
            "node_id": "IC_kwDOBRZfO845X__f",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/962592735/reactions"
            },
            "updated_at": "2021-11-07T11:21:56Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/962592735",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
                "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
                "followers_url": "https://api.github.com/users/weiji14/followers",
                "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
                "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/weiji14",
                "id": 23487320,
                "login": "weiji14",
                "node_id": "MDQ6VXNlcjIzNDg3MzIw",
                "organizations_url": "https://api.github.com/users/weiji14/orgs",
                "received_events_url": "https://api.github.com/users/weiji14/received_events",
                "repos_url": "https://api.github.com/users/weiji14/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/weiji14"
            }
        }
    ],
    "comments_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129/comments",
    "created_at": "2021-03-26T18:04:32Z",
    "events_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129/events",
    "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1129",
    "id": 842191607,
    "labels": [
        {
            "color": "ee0701",
            "default": true,
            "description": "Something isn't working",
            "id": 563309827,
            "name": "bug",
            "node_id": "MDU6TGFiZWw1NjMzMDk4Mjc=",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/labels/bug"
        }
    ],
    "labels_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129/labels{/name}",
    "locked": false,
    "milestone": {
        "closed_at": null,
        "closed_issues": 164,
        "created_at": "2022-06-12T13:10:39Z",
        "creator": {
            "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
            "events_url": "https://api.github.com/users/seisman/events{/privacy}",
            "followers_url": "https://api.github.com/users/seisman/followers",
            "following_url": "https://api.github.com/users/seisman/following{/other_user}",
            "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/seisman",
            "id": 3974108,
            "login": "seisman",
            "node_id": "MDQ6VXNlcjM5NzQxMDg=",
            "organizations_url": "https://api.github.com/users/seisman/orgs",
            "received_events_url": "https://api.github.com/users/seisman/received_events",
            "repos_url": "https://api.github.com/users/seisman/repos",
            "site_admin": false,
            "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
            "type": "User",
            "url": "https://api.github.com/users/seisman"
        },
        "description": "",
        "due_on": "2022-11-01T07:00:00Z",
        "html_url": "https://github.com/GenericMappingTools/pygmt/milestone/14",
        "id": 8073518,
        "labels_url": "https://api.github.com/repos/GenericMappingTools/pygmt/milestones/14/labels",
        "node_id": "MI_kwDOBRZfO84AezEu",
        "number": 14,
        "open_issues": 62,
        "state": "open",
        "title": "0.8.0",
        "updated_at": "2022-11-07T11:23:55Z",
        "url": "https://api.github.com/repos/GenericMappingTools/pygmt/milestones/14"
    },
    "node_id": "MDU6SXNzdWU4NDIxOTE2MDc=",
    "number": 1129,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 0,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 0,
        "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129/reactions"
    },
    "repository_url": "https://api.github.com/repos/GenericMappingTools/pygmt",
    "state": "closed",
    "state_reason": "completed",
    "timeline_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129/timeline",
    "title": "pygmt.Figure.meca offset not working",
    "updated_at": "2022-07-03T15:56:18Z",
    "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1129",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/81437398?v=4",
        "events_url": "https://api.github.com/users/rdcorona/events{/privacy}",
        "followers_url": "https://api.github.com/users/rdcorona/followers",
        "following_url": "https://api.github.com/users/rdcorona/following{/other_user}",
        "gists_url": "https://api.github.com/users/rdcorona/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/rdcorona",
        "id": 81437398,
        "login": "rdcorona",
        "node_id": "MDQ6VXNlcjgxNDM3Mzk4",
        "organizations_url": "https://api.github.com/users/rdcorona/orgs",
        "received_events_url": "https://api.github.com/users/rdcorona/received_events",
        "repos_url": "https://api.github.com/users/rdcorona/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/rdcorona/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/rdcorona/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/rdcorona"
    }
}